src/app/ReadClient.cpp

```diff
@@ -34,47 +34,10 @@
 namespace chip {
 namespace app {
 
-/**
- * @brief The default resubscribe policy will pick a random timeslot
- * with millisecond resolution over an ever increasing window,
- * following a fibonacci sequence up to CHIP_RESUBSCRIBE_MAX_FIBONACCI_STEP_INDEX,
- * Average of the randomized wait time past the CHIP_RESUBSCRIBE_MAX_FIBONACCI_STEP_INDEX
- * will be around one hour.
- * When the retry count resets to 0, the sequence starts from the beginning again.
- */
-static void DefaultResubscribePolicy(uint32_t aNumCumulativeRetries, uint32_t & aNextSubscriptionIntervalMsec,
-                                     bool & aShouldResubscribe)
-{
-    uint32_t maxWaitTimeInMsec = 0;
-    uint32_t waitTimeInMsec    = 0;
-    uint32_t minWaitTimeInMsec = 0;
-
-    if (aNumCumulativeRetries <= CHIP_RESUBSCRIBE_MAX_FIBONACCI_STEP_INDEX)
-    {
-        maxWaitTimeInMsec = GetFibonacciForIndex(aNumCumulativeRetries) * CHIP_RESUBSCRIBE_WAIT_TIME_MULTIPLIER_MS;
-    }
-    else
-    {
-        maxWaitTimeInMsec = CHIP_RESUBSCRIBE_MAX_RETRY_WAIT_INTERVAL_MS;
-    }
-
-    if (maxWaitTimeInMsec != 0)
-    {
-        minWaitTimeInMsec = (CHIP_RESUBSCRIBE_MIN_WAIT_TIME_INTERVAL_PERCENT_PER_STEP * maxWaitTimeInMsec) / 100;
-        waitTimeInMsec    = minWaitTimeInMsec + (Crypto::GetRandU32() % (maxWaitTimeInMsec - minWaitTimeInMsec));
-    }
-
-    aNextSubscriptionIntervalMsec = waitTimeInMsec;
-    aShouldResubscribe            = true;
-    ChipLogProgress(DataManagement,
-                    "Computing Resubscribe policy: attempts %" PRIu32 ", max wait time %" PRIu32 " ms, selected wait time %" PRIu32
-                    " ms",
-                    aNumCumulativeRetries, maxWaitTimeInMsec, waitTimeInMsec);
-}
-
 ReadClient::ReadClient(InteractionModelEngine * apImEngine, Messaging::ExchangeManager * apExchangeMgr, Callback & apCallback,
```

Comment:
It is still effectively a pointer, but one that cannot be null right?

I don't believe there is a real convention in the SDK around naming here, and since this existed before this PR, I'm inclined to keep it in the way it is for now.